### PR TITLE
fix(settings): repair 7 PVSettings test failures; make scaling migration order-independent

### LINKED ISSUE #3651: PVSettings: 7 pre-existing test failures exposed by spm-validate.sh fix
## Summary

`Scripts/spm-validate.sh` has been exiting after its **first passing module** for its entire life, so the "Build & test SPM leaf modules" job never validated 4 of its 5 modules. With the script fixed (#3650, commit `673b5350fd`), `PVSettings` runs for the first time and fails **7 assertions**.

These are pre-existing and unrelated to #3650 (that branch touches zero PVSettings files). Filing separately so the importer PR isn't scope-crept.

## Why they were invisible

```bash
bash -euo pipefail -c 'P=0; ((P++)); echo SURVIVED'   # prints nothing, exit 1
```

`((VAR++))` post-increment evaluates to the OLD value; with `VAR=0` the arithmetic command's result is zero, which bash reports as exit status 1, and `set -e` kills the script. The job logged `PASS: PVLogging` then died — reporting failure while everything it ran passed. Every `SPM Module Validation` run on `develop` has been red since at least 2026-05-27 for this reason.

## The 7 failures

Run `cd PVSettings && swift test` to reproduce.

**1. Stale expectation — `LightGunCrosshairStyle has three cases`** (`PVSettingsTests.swift:1613`)
```
(LightGunCrosshairStyle.allCases.count → 4) == 3
```
The enum gained a 4th case (`reticle`) in #3497 / `308365688e`; the test was never updated. Fix: assert 4, and ideally assert the specific case set so additions are caught explicitly.

**2. Swift optional-vs-case ambiguity — `None parses from rawValue`** (`PVSettingsTests.swift:566`)
```
(parsed → None) == (.none → nil)
```
`parsed` is `MetalFilterModeOption?`, so the bare `.none` in `#expect(parsed == .none)` resolves to `Optional.none` (nil), **not** `MetalFilterModeOption.none`. The test can never pass. Fix: `#expect(parsed == MetalFilterModeOption.none)`.

**3–5. ScalingMode migration tests** (`PVSettingsTests.swift:1848`, `:1859`, `:1870`)
```
(ud.string(forKey: "scalingMode") → "aspectFit") == "integerScale"   // and nativeResolution
(ud.object(forKey: "scalingMode") → "aspectFit") == nil
```
These use `UserDefaults(suiteName:)` for isolation, but a suite's **search list still includes the standard/registration domains**, so a `scalingMode` value written by another test (Swift Testing runs suites in parallel) is visible through it — and `migrateScalingModeIfNeeded` then sees a non-nil value and correctly skips migrating.

**6. `effectiveCrosshairStyle falls back to global when no per-game override`** (`:1747`)
```
(Defaults.effectiveCrosshairStyle(forGameMD5: md5) → .crosshair) == .dot
```
Same class: depends on the global crosshair default not having been mutated by a concurrently-running test.

**7. `ControllerLayoutSettings / Default is empty dictionary`** (`:1489`)
```
(Defaults[.controllerLayoutVariantsBySystem] → ["com.provenance.genesis": "genesis-6btn"]).isEmpty → false
```
Reads standard `Defaults` and asserts a pristine value; another test writes the key first.

## Suggested fixes

- **1 and 2** are straightforward test corrections.
- **3–7** are test-isolation problems, not product bugs. Options: mark the affected suites `.serialized`, give each test a genuinely private `UserDefaults` (unique suite name + `removePersistentDomain` in both setup and teardown, and read back through that same instance only), or have the tests explicitly clear the keys they assert on before asserting.

Worth confirming per-test whether the product behaviour is right before changing an expectation — #1 in particular is a case where the *test* is stale, but the same shape could hide a real regression.

## Note

Since branch protection requires no status checks (`contexts: []`), this red job doesn't block merges — which is likely why it stayed red for months. Fixing these would make the job meaningful again.

### PR DESCRIPTION
**Pull request by JoeMatt**
Closes #3651.

## Correction to an earlier version of this description

An earlier revision of this PR claimed the legacy scaling-mode migration "never ran for any user." **That was wrong** — I've corrected it below. The production path was fine; the tests were not.

## Summary

The 7 PVSettings failures exposed when `spm-validate.sh` was fixed (#3650) were three separate problems, all test-side.

### 1. Stale test — `LightGunCrosshairStyle`

The enum gained a 4th case (`reticle`) in #3497 / `308365688e`; the test still asserted 3. Now asserts the exact case set, so renames are caught too, not just additions.

### 2. Broken test — `MetalFilterModeOption` "None parses"

```swift
#expect(parsed == .none)
```

`parsed` is `MetalFilterModeOption?`, so the bare `.none` resolves to `Optional.none` (nil) rather than `MetalFilterModeOption.none`. This could never pass regardless of behaviour. Now names the type explicitly.

### 3. Test isolation

`ControllerLayoutSettings`, `Light Gun Settings` and `ScalingMode` all mutate shared `Defaults` but weren't `.serialized`, unlike every other Defaults-touching suite in the same file (`SkinMode Defaults Migration`, `Mouse Defaults Keys`, `Gyro Mouse Defaults`, …). Marked accordingly.

## The ScalingMode migration change — and why it is *not* a user-facing bug fix

The three migration tests failed for a subtler reason. `migrateScalingModeIfNeeded` guarded on:

```swift
guard userDefaults.object(forKey: "scalingMode") == nil else { return }
```

`object(forKey:)` resolves through the **registration domain**, which `Defaults` populates for every declared `Key` (`Defaults.swift:140`, `suite.register(defaults:)`) and which is process-wide. A test calling this function directly — after any other test had already forced `Defaults.Keys.scalingMode` to initialise — always saw `"aspectFit"` and skipped the migration.

**Production was not affected.** The only production caller is inside the lazy initialiser of the key itself:

```swift
static let scalingMode: Key<ScalingMode> = {
    migrateScalingModeIfNeeded()                       // runs FIRST…
    return Key<ScalingMode>("scalingMode", default: .aspectFit)   // …which registers the default
}()
```

Migration runs *before* the default is registered, so the guard correctly saw `nil` and the migration did run for upgrading users.

This PR changes the guard to ask the **persistent** domain whether the user ever explicitly chose a mode, and adds an injectable `domainName`:

- correctness no longer depends on that initialisation ordering (a subsequent refactor moving the call would silently break migration),
- the function becomes testable in isolation.

Treat it as robustness + testability, not a shipped-bug fix.

## Verification

```
swift test --package-path PVSettings
→ Test run with 223 tests in 30 suites passed
```

Previously 7 failures. This is the first time this module's tests have passed — they had never actually run, because `spm-validate.sh` exited after its first module until #3650 fixed it.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: PVSettings/Tests/PVSettingsTests/PVSettingsTests.swift [hunk 1]
```diff
@@ -1840,34 +1846,40 @@ struct ScalingModeTests {
 
     @Test("migration — maps integerScaleEnabled=true to .integerScale")
     func migrationIntegerScale() {
-        let ud = UserDefaults(suiteName: "test.scalingmode.integer")!
+        let suite = "test.scalingmode.integer"
```
[the remark this answers is not in the document]
Comment (JoeMatt) [reply PVSettings/Tests/PVSettingsTests/PVSettingsTests.swift:1852]: Fixed in 1a3753c6b5 — it now passes `domainName: suite` and asserts against `ud.persistentDomain(forName: suite)?["scalingMode"]`, so it actually exercises the guard it names. Good catch: it had been passing by luck (no legacy flags set in that suite meant the migration wrote nothing either way).

File: PVSettings/Sources/PVSettings/Settings/Model/PVSettingsModel.swift [hunk 2]
```diff
@@ -889,9 +889,27 @@ internal func makeMigratingBoolKey(_ primaryKey: String, legacyKey: String, defa
...
+/// store where nothing had ever been written. The previous
+/// `guard userDefaults.object(forKey: scalingKey) == nil` was therefore ALWAYS false and
+/// this migration never ran for anyone: users upgrading with `integerScaleEnabled = true`
+/// silently lost the setting.
+internal func migrateScalingModeIfNeeded(userDefaults: UserDefaults = .standard,
+                                         domainName: String = Bundle.main.bundleIdentifier ?? "") {
     let scalingKey = "scalingMode"
-    guard userDefaults.object(forKey: scalingKey) == nil else { return }
+    let explicitlySetByUser = userDefaults.persistentDomain(forName: domainName)?[scalingKey] != nil
+    guard !explicitlySetByUser else { return }
```
[the remark this answers is not in the document]
Comment (JoeMatt) [reply PVSettings/Sources/PVSettings/Settings/Model/PVSettingsModel.swift:920]: Agreed, and taken further than suggested. In 1a3753c6b5 `domainName` is now `String?` defaulting to `nil`, and **nil keeps the original `object(forKey:)` behaviour** — so production is byte-identical to before this PR, and the parameter exists purely so tests can target their own suite. That removes the risk you describe entirely rather than just narrowing it: no bundle-identifier derivation happens at all.

Worth adding that the premise of my original description was also wrong — I'd claimed the migration never ran for anyone. It did: the only production caller sits inside the lazy initialiser of `Defaults.Keys.scalingMode` and runs *before* `Key(...)` registers the default, so `object(forKey:)` correctly saw nil. The PR description has been corrected.

File: PVSettings/Sources/PVSettings/Settings/Model/PVSettingsModel.swift [hunk 3]
```diff
@@ -889,9 +889,27 @@ internal func makeMigratingBoolKey(_ primaryKey: String, legacyKey: String, defa
...
+///
+/// - Parameters:
+///   - userDefaults: Store to migrate.
+///   - domainName: Persistent domain to inspect for an explicitly-set value. Defaults to
+///     the app's own domain; tests pass their suite name.
+///
+/// IMPORTANT: "has the user set a value?" must be answered against the PERSISTENT domain,
+/// not `object(forKey:)`. `Defaults` registers every declared Key's default value into the
+/// process-wide registration domain (Defaults.swift:140 `suite.register(defaults:)`), and
+/// `object(forKey:)` resolves through that domain — so it returned "aspectFit" even for a
```
[the remark this answers is not in the document]
Comment (JoeMatt) [reply PVSettings/Sources/PVSettings/Settings/Model/PVSettingsModel.swift]: Agreed — removed in 1a3753c6b5. The explanation now stays at the API level (that `Defaults` registers each declared Key's default into the process-wide registration domain) without pinning a line number in an external dependency.

File: PVSettings/Sources/PVSettings/Settings/Model/PVSettingsModel.swift [hunk 4]
```diff
@@ -889,9 +889,36 @@ internal func makeMigratingBoolKey(_ primaryKey: String, legacyKey: String, defa
 /// - `integerScaleEnabled = true` maps to `.integerScale` (takes precedence).
 /// - `nativeScaleEnabled = true` maps to `.nativeResolution`.
 /// - Both false (or not set) maps to the default `.aspectFit`.
-internal func migrateScalingModeIfNeeded(userDefaults: UserDefaults = .standard) {
+/// Migrates the legacy `integerScaleEnabled` / `nativeScaleEnabled` booleans to the
+/// unified `scalingMode` key, but only when the user has never chosen a scaling mode.
+///
```
[the remark this answers is not in the document]
Comment (JoeMatt) [reply PVSettings/Sources/PVSettings/Settings/Model/PVSettingsModel.swift]: Fixed — merged into one block that keeps both the legacy-flag mapping table and the 'only when the user has never chosen a mode' condition.

File: .github/workflows/agent-validation.yml [hunk 5]
```diff
@@ -199,7 +199,10 @@ jobs:
         run: |
           set -o pipefail
           cd ${{ matrix.module }}
-          swift test 2>&1 | tail -40
+          # --no-parallel: PVSettings (and other Defaults-backed suites) share
+          # process-global UserDefaults; Swift Testing's default concurrent suite
+          # execution made them flake. See Scripts/spm-validate.sh for detail.
+          swift test --no-parallel 2>&1 | tail -40
```
[the remark this answers is not in the document]
Comment (JoeMatt) [reply .github/workflows/agent-validation.yml:205]: Noting it here as the rule asks. This PR changes `.github/workflows/agent-validation.yml` in one place: the SPM test step now runs `swift test --no-parallel`.

Reason: PVSettings' 223 tests all read and write `UserDefaults.standard` through `Defaults`, and Swift Testing runs suites concurrently by default. `.serialized` only orders tests *within* a suite, so cross-suite writes still raced — `imageSmoothing` failed 1 run in 3. With `--no-parallel` it is 4/4 clean, and the suite runs in ~0.04s so serial execution costs nothing.

The matching change is in `Scripts/spm-validate.sh`, so local and CI behave identically. No permissions, triggers or job structure were touched.
